python/ray/dag/compiled_dag_node.py

```diff
@@ -150,10 +169,25 @@ def __init__(self, idx: int, dag_node: "ray.dag.DAGNode"):
         """
         self.idx = idx
         self.dag_node = dag_node
+        self.arg_idx_to_tensor_meta: Dict[int, Dict[str, Any]] = {}
 
         self.downstream_node_idxs = set()
         self.output_channel = None
 
+        self.output_wrapper_fn = None
+        if self.dag_node.type_hint is not None:
+            if isinstance(self.dag_node.type_hint, TorchTensorType):
```

Comment:
wonder if we can detect torch tensor automatically here? and if it is on gpu we just wrap it to this wrapper?

Reply:
I wanted to preserve existing serialization methods for torch.Tensor since this acts a bit differently from normal serialization; it automatically sets the device on the receiving end.

There is probably a nicer way to do this, but for now this is the easiest and probably doesn't affect performance.